### PR TITLE
feat: [go-feature-flag] Support flag metadata

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR
- Support flag metadata in the java provider for GO Feature Flag

### CODE REVIEW

**Review by toddbaert — APPROVED** [review]
Nice to have!

File: providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java [hunk 1]
```diff
@@ -12,35 +9,23 @@
...
-import dev.openfeature.sdk.EvaluationContext;
-import dev.openfeature.sdk.FeatureProvider;
-import dev.openfeature.sdk.Hook;
-import dev.openfeature.sdk.Metadata;
-import dev.openfeature.sdk.MutableStructure;
-import dev.openfeature.sdk.ProviderEvaluation;
-import dev.openfeature.sdk.Reason;
-import dev.openfeature.sdk.Structure;
-import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.*;
```
Comment (Kavindu-Dodan) [inline providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java]: minor - personally prefer not to have wildcard imprts

File: providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java [hunk 2]
```diff
@@ -12,35 +9,23 @@
...
 import dev.openfeature.sdk.exceptions.TypeMismatchError;
-import okhttp3.ConnectionPool;
-import okhttp3.HttpUrl;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
-import okhttp3.ResponseBody;
+import okhttp3.*;
```
Comment (Kavindu-Dodan) [inline providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java]: minor - personally prefer not to have wildcard imprts

File: providers/go-feature-flag/src/test/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProviderTest.java [hunk 3]
```diff
@@ -7,18 +7,14 @@
 import java.util.Arrays;
 import java.util.List;
 
+import dev.openfeature.sdk.*;
```
Comment (Kavindu-Dodan) [inline providers/go-feature-flag/src/test/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProviderTest.java]: wildcard imports

**Review by Kavindu-Dodan — APPROVED** [review]

File: providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java [hunk 4]
```diff
@@ -256,6 +259,32 @@ private ErrorCode mapErrorCode(String errorCode) {
...
+    /**
+     * convertFlagMetadata is converting the flagMetadata object received from the server
+     * to an ImmutableMetadata format known by Open Feature.
+     *
+     * @param flagMetadata - metadata received from the server
+     * @return a converted metadata object.
+     */
+    private ImmutableMetadata convertFlagMetadata(Map<String, Object> flagMetadata) {
+        ImmutableMetadata.ImmutableMetadataBuilder builder = ImmutableMetadata.builder();
+        flagMetadata.forEach((k, v) -> {
```
Comment (itchyny) [inline providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java:271]: We're facing `NullPointerException` on introducing the flag sdk into our server. The `metadata` should be optional.

Comment (Gregory-Berkman-Imprivata) [reply providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java:271]: I am seeing this as well

Comment (toddbaert) [reply providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java:271]: FYI @thomaspoignant

**Pull request merged**